### ISSUE TITLE
[Bug]: rsync 3.2.4 without support for ipv6

### ISSUE THREAD

**Issue by RainGather** [termux/termux-packages#11468]
### Problem description

Rsync is built without support for ipv6.

### What steps will reproduce the bug?

Steps to reproduce

rsync --version

rsync  version 3.2.4  protocol version 31
...
Capabilities:
    ... no ipv6 ...

### What is the expected behavior?

rsync can use ipv6 like old version.

### System information

termux-info:

```
Termux Variables:
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=21668
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://mirrors.tuna.tsinghua.edu.cn/termux/apt/termux-main stable main
Updatable packages:
All packages up to date
termux-tools version:
1.24
Android version:
11
Kernel build information:
Linux localhost 4.14.186-ga715ffbf0569 #1 SMP PREEMPT Mon Feb 21 22:23:39 CST 2022 aarch64 Android
Device manufacturer:
Xiaomi
Device model:
21091116C
```

**Comment by 2096779623** [comment]
```
~ $ rsync -6 rsync://mirror.bardia.tech/termux              mmmmm                   #    "               m                  #
 #    #  mmm   m mm   mmm#  mmm     mmm     mm#mm   mmm    mmm   # mm                                                  #mmmm" "   #  #"  " #" "#    #    "   #      #    #"  #  #"  "  #"  #
 #    # m"""#  #     #   #    #    m"""#      #    #""""  #      #   #
 #mmmm" "mm"#  #     "#m##  mm#mm  "mm"#  #   "mm  "#mm"  "#mm"  #   #

drwxr-xr-x          4,096 2022/02/16 01:17:33 .
drwxr-xr-x          4,096 2022/02/16 01:17:29 game-packages-24
drwxr-xr-x          4,096 2022/02/16 01:17:30 science-packages-24
drwxr-xr-x          4,096 2022/02/16 01:17:31 termux-packages-24
drwxr-xr-x          4,096 2022/02/16 01:17:32 termux-root-packages-24
drwxr-xr-x          4,096 2022/02/16 01:17:33 unstable-packages
drwxr-xr-x          4,096 2022/02/16 01:17:33 x11-packages
```

**Comment by RainGather** [comment]
When I use ssh:
```shell
ssh -p 8036 user@myipv6host.top
```
ssh Everything OK.

But when use rsync:
```shell
rsync -auv --progress -e "ssh -p 8036" user@myipv6host.top:/test/ /sdcard/test/
```
```shell
ssh: Could not resolve hostname myipv6host.top: No address associated with hostname
rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(228) [sender=3.2.4]
```

And I use IP direct:
```shell
rsync -auv --progress -e "ssh -p 8036" user@[2409:8a28:1a9c:b5f0:4402:a0ff:fe07:a9e2]:/test/ /sdcard/test/
```
Error:
```shell
ssh: Could not resolve hostname 2409:8a28:1a9c:b5f0:4402:a0ff:fe07:a9e2: No address associated with hostname                                                  
rsync: connection unexpectedly closed (0 bytes received so far) [sender]                           
rsync error: unexplained error (code 255) at io.c(228) [sender=3.2.4]
```

And everything is OK until I `pkg upgrade` one day.

**Comment by tomty89** [comment]
I could reproduce it and I confirmed that it's fixed with #11469 on my device. Please check again when 3.2.4-1 is available on the repo. Closing.
